PVSSpringBoot/src/test/java/com/example/PVSSpringBoot/ControllerPackage/ControllerTest.java

```diff
@@ -0,0 +1,165 @@
+import com.example.PVSSpringBoot.repositories.UsersRepo;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import com.example.PVSSpringBoot.ControllerPackage.Controller;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+
+@WebMvcTest(Controller.class)
+@ContextConfiguration(classes = Controller.class)
+@AutoConfigureMockMvc(addFilters = false)
+@WithMockUser
+class ControllerTest {
+    @Autowired
+    MockMvc mockMvc;
+
+    @MockBean
+    RequestService requestService;
+    @MockBean
+    UsersRepo usersRepo;
+
+    @Test
+    void addNewProductCorrect() throws Exception {
+        ProductFrontBuilder productBuilder = new ProductFrontBuilder();
+        ProductFront product = productBuilder.get();
+        Mockito.when(requestService.addNewProduct(any())).thenReturn("Added to database...");
+        mockMvc.perform(MockMvcRequestBuilders
+                .post("/api/addNewProduct")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("{\n" +
+                 "   \"name\": \"foodStrong1221111\",\n" +
+                 "    \"brandName\": \"petpet8\",\n" +
+                 "    \"description\": \"fooooooood\",\n" +
+                 "    \"price\": \"1.66\",\n" +
+                 "    \"categoryName\": \"food\",\n" +
+                 "    \"targetAnimal\": \"cat\",\n" +
+                 "    \"userEmail\": \"abdo2@gmail.com\"\n" +
+                 "}"
+                ))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.content().string("Added to database..."));
+                Mockito.verify(requestService,Mockito.times(1)).addNewProduct(any());
+    }
+    @Test
+    void addNewProductFailed() throws Exception {
+        ProductFrontBuilder productBuilder = new ProductFrontBuilder();
+        ProductFront product = productBuilder.get();
+        Mockito.when(requestService.addNewProduct(any())).thenReturn("The email of user is wrong");
+        mockMvc.perform(MockMvcRequestBuilders
+                        .post("/api/addNewProduct")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\n" +
```

Comment:
this part of code repeated and could be replaced with function that takes parameters of url , content 
for example
private ResultActions performPostRequest(String url, String content) throws Exception {
    return mockMvc.perform(MockMvcRequestBuilders
            .post(url)
            .contentType(MediaType.APPLICATION_JSON)
            .content(content));
}